### PR TITLE
feat(build): add support for building on arm and x86

### PR DESCRIPTION
**Pull request by wangzihao3**
Related to: openebs/openebs#1295

- modify build_image.sh to decide which arch is used

Signed-off-by: wangzihao <wangzihao18@huawei.com>

### CODE REVIEW

File: build_image.sh [hunk 1]
```diff
@@ -2,14 +2,28 @@
 set -e
 
 pwd
+
+# Determine the arch/os we're building for
+ARCH=$(uname -m)
+OS=$(uname)
```
Comment (mynktl) [inline build_image.sh]: Unused variable.

**Review by mynktl — APPROVED** [review]

File: build_image.sh (hunk 1 above)
Comment (wangzihao3) [reply build_image.sh]: oh yes, I will delete it.

File: docker/Dockerfile.arm64 [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+RUN touch /usr/local/etc/bkpistgt/logfile
+
+COPY entrypoint-istgtimage.sh /usr/local/bin/
+RUN chmod +x /usr/local/bin/entrypoint-istgtimage.sh
+
+ARG BUILD_DATE
+LABEL org.label-schema.name="cstor"
+LABEL org.label-schema.description="OpenEBS cstor"
+LABEL org.label-schema.url="http://www.openebs.io/"
+LABEL org.label-schema.vcs-url="https://github.com/openebs/cstor"
```
Comment (pawanpraka1) [inline docker/Dockerfile.arm64]: it should be https://github.com/openebs/istgt.

File: docker/Dockerfile.arm64 [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+COPY istgt istgtcontrol /usr/local/bin/
+COPY istgt.conf istgtcontrol.conf /usr/local/etc/bkpistgt/
+RUN touch /usr/local/etc/bkpistgt/auth.conf
+RUN touch /usr/local/etc/bkpistgt/logfile
+
+COPY entrypoint-istgtimage.sh /usr/local/bin/
+RUN chmod +x /usr/local/bin/entrypoint-istgtimage.sh
+
+ARG BUILD_DATE
+LABEL org.label-schema.name="cstor"
```
Comment (pawanpraka1) [inline docker/Dockerfile.arm64]: istgt

File: docker/Dockerfile.arm64 [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+LABEL org.label-schema.description="OpenEBS cstor"
```
Comment (pawanpraka1) [inline docker/Dockerfile.arm64]: OpenEBS istgt

**Comment by wangzihao3** [comment]
Thanks @pawanpraka1 . I will update them. There are the same issue in Dockerfile. Should I update them both ?

**Comment by pawanpraka1** [comment]
yes, please go ahead. Fix the existing Dockerfile also.

File: docker/entrypoint-istgtimage.sh [hunk 5]
```diff
@@ -20,6 +20,7 @@ touch /usr/local/etc/istgt/logfile
 export externalIP=0.0.0.0
 service rsyslog start
 
-export LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libjemalloc.so
+ARCH=$(uname -m)
```
Comment (pawanpraka1) [inline docker/entrypoint-istgtimage.sh:23]: unused? What's the purpose of this variable?

Comment (wangzihao3) [reply docker/entrypoint-istgtimage.sh:23]: I used to define a OS=$(uname) variable which is unused. This one($ARCH) is used to define the path of LD_PRELOAD. export LD_PRELOAD=/usr/lib/${ARCH}-linux-gnu/libjemalloc.so

Comment (pawanpraka1) [reply docker/entrypoint-istgtimage.sh:23]: I missed that, thanks. Looks good.

**Review by pawanpraka1 — APPROVED** [review]
looks good.

**Pull request merged**